xwayland/xwayland.c

```diff
@@ -86,10 +90,12 @@ static void exec_xwayland(struct wlr_xwayland *wlr_xwayland) {
 		_exit(EXIT_FAILURE);
 	}
 
+#ifdef __linux__
 	if (clearenv()) {
 		wlr_log_errno(L_ERROR, "clearenv failed");
 		_exit(EXIT_FAILURE);
 	}
+#endif
```

Comment:
You can't do this on FreeBSD? We should put in some other kind of mechanism, then.

Reply:
> There's the environ linking issue (undefined symbol).

I removed the `environ = NULL` because of this. I guess removing the `no-undefined` linker flag is also an option…

Reply:
No, no, don't add that flag o_o

Simply figure out another way to clear the environment on BSD.

Reply:
There's no way other than `environ = NULL`.

And `environ` is in `crt1.o` instead of `libc.so` — which means the linker thinks it's undefined. https://bugs.webkit.org/show_bug.cgi?id=138420

Why is the environment cleared for Xwayland?

Reply:
I'm not sure. Maybe @martinetd or @emersion knows.

Reply:
I don't know why the environment is cleared. But [wlc does it too](https://github.com/Cloudef/wlc/blob/master/src/xwayland/xwayland.c#L295), without having to ifdef linux.

Reply:
Oh, right, check out where that clearenv comes from https://github.com/Cloudef/wlc/blob/328edc089915aeb5cf26e5fc0e81b307ec707d62/src/internal.h#L395-L401 :)

CMake does not add `-Wl,--no-undefined` by default like Meson does, so wlc builds fine.

Reply:
> check out where that clearenv comes from https://github.com/Cloudef/wlc/blob/328edc089915aeb5cf26e5fc0e81b307ec707d62/src/internal.h#L395-L401 
Yeah, I did that...

Reply:
Yeah I've added that clearenv because wlc (and weston iirc) do it ; I think we could do without if we clear some key values instead but clearenv() sounds a lot less painful to me... My initial implementation used execvep but that was a GNU extension, I guess we could try to execve for every component of the PATH instead but that's a bit unsightly... There's not the problem you link from execve because we don't need to retrieve the whole previous env.

That said, I'm tempted to say it's easiest to do like wlc: define environ as extern and call it a day (either clear directly or redefine clearenv like them)

Reply:
Ah, I should learn to read before replying ; extern won't work with -Wl,--no-undefined?

That's gonna be tricky... Can try execve then, I like the idea of messing with warnings or linked objects quite less!